core/src/main/scala/chisel3/Data.scala

```diff
@@ -520,63 +520,81 @@ abstract class Data extends HasId with NamedComponent with SourceInfoDoc {
 
   /** Whether this Data has the same model ("data type") as that Data.
     * Data subtypes should overload this with checks against their own type.
+    * @param that the Data to check for type equivalence against.
+    * @param strictProbeInfo whether probe info (including its RW-ness and Color) must match
     */
   private[chisel3] final def typeEquivalent(
-    that: Data
-  ): Boolean = findFirstTypeMismatch(that, strictTypes = true, strictWidths = true).isEmpty
+    that:            Data,
+    strictProbeInfo: Boolean = true
+  ): Boolean =
+    findFirstTypeMismatch(that, strictTypes = true, strictWidths = true, strictProbeInfo = strictProbeInfo).isEmpty
 
   /** Find and report any type mismatches
     *
     * @param that Data being compared to this
     * @param strictTypes Does class of Bundles or Records need to match? Inverse of "structural".
     * @param strictWidths do widths need to match?
+    * @param strictProbeInfo does probe info need to match (includes RW and Color)
     * @return None if types are equivalent, Some String reporting the first mismatch if not
     */
   private[chisel3] final def findFirstTypeMismatch(
-    that:         Data,
-    strictTypes:  Boolean,
-    strictWidths: Boolean
+    that:            Data,
+    strictTypes:     Boolean,
+    strictWidths:    Boolean,
+    strictProbeInfo: Boolean
   ): Option[String] = {
+
+    def checkProbeInfo(left: Data, right: Data): Option[String] =
+      Option.when(strictProbeInfo && (left.probeInfo != right.probeInfo)) {
+        def probeInfoStr(info: Option[ProbeInfo]) = info.map { info =>
+          s"Some(writeable=${info.writable}, color=${info.color})"
+        }.getOrElse("None")
+        s": Left (${left} with probeInfo: ${probeInfoStr(left.probeInfo)}) and Right ($right with probeInfo: (${probeInfoStr(right.probeInfo)}) have different probeInfo."
```

Comment:
```suggestion
        s": Left ($left with probeInfo: ${probeInfoStr(left.probeInfo)}) and Right ($right with probeInfo: (${probeInfoStr(right.probeInfo)})) have different probeInfo."
```
Add missing parentheses and remove unneeded curly braces.